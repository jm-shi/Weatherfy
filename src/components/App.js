var React = require('react');
var Home = require('./Home');
var Forecast = require('./Forecast');
var ReactRouter = require('react-router-dom');
var BrowserRouter = ReactRouter.BrowserRouter;
var Route = ReactRouter.Route;
var Switch = ReactRouter.Switch;

class App extends React.Component {
    render() {
        return (
            <BrowserRouter>
                <div className='container'>
                    <Switch>
                        <Route exact path='/' component={Home} />
                        <Route path='/forecast' component={Forecast} />
                        <Route render={function() {
                            return <div id='notFound'>
                                    <div id='notFoundHeader'>404</div>
                                    <div id='notFoundText'>Page Not Found</div>
                                  </div> }} />
                    </Switch>
                </div>
            </BrowserRouter>
        );
    }
}

module.exports = App;