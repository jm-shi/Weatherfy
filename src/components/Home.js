var React = require('react');
var Form = require('./Form');

class Home extends React.Component {
    render() {
        return (
            <Form />
        );
    }
}

module.exports = Home;