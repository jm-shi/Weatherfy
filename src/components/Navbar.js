const React = require('react');
const Searchbar = require('./Searchbar');
const NavLink = require('react-router-dom').NavLink;

class Navbar extends React.Component {
    render() {
        return (
            <div className='header'>
                <NavLink style={{'fontSize': '5em'}} exact to='/'>WEATHER</NavLink>
                <div style={{ 'marginTop': '20px' }}>
                    <Searchbar />
                </div>
            </div>
        );
    }
}

module.exports = Navbar;