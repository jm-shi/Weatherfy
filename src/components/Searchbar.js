const React = require('react');
const Link = require('react-router-dom').Link;
const api = require('../utils/api');

class Searchbar extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            zipcode: ''
        }

        this.setZipcode = this.setZipcode.bind(this);
        this.checkForm = this.checkForm.bind(this);
    }

    setZipcode(e) {
        var value = e.target.value;
        this.setState(() => ({zipcode: value}))
    }

    checkForm(e) {
        if (this.state.zipcode.length !== 5) {
            e.preventDefault();
            alert("Enter a 5-digit zipcode.");
        } 
    }

    render() {
        const { zipcode } = this.state;

        return (
            <div style={{'display': 'flex'}}>
                <input
                    id='searchBar'
                    type='text'
                    placeholder='Enter a zipcode' 
                    value={zipcode} 
                    onChange={this.setZipcode} />


                <Link className='button' 
                    to={
                        {
                            pathname: `/forecast/${zipcode}`,
                            zipcode: zipcode
                        }
                    }
                    onClick={this.checkForm} >
                    Search
                </Link>
            </div>
        );
    }
}

module.exports = Searchbar;
