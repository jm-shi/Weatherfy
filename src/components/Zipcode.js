const React = require('react');
const Searchbar = require('./Searchbar');
const Link = require('react-router-dom').Link;
const api = require('../utils/api');

class Zipcode extends React.Component {
    render() {
        return (
            <div>
                <form className='form'>
                    <div className='formContainer'>
                        <label id='searchLabel' htmlFor='searchBar'>Get weather by zipcode</label>
                        <Searchbar />
                    </div> 
                </form>
            </div>
        )
    }
}

module.exports = Zipcode;