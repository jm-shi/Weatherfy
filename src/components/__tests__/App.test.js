import React from 'react';
import App from '../App.js';
import Home from '../Home.js';
import Navbar from '../Navbar.js';
import Searchbar from '../Searchbar.js';
import Zipcode from '../Zipcode.js';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('<App />', () => {
    it('should render correctly', () => {
      const output = shallow(<App />);
      expect(shallowToJson(output)).toMatchSnapshot();
    });
});

describe('<Home />', () => {
    it('should render correctly', () => {
      const output = shallow(<Home />);
      expect(shallowToJson(output)).toMatchSnapshot();
    });
});

describe('<Navbar />', () => {
    it('should render correctly', () => {
      const output = shallow(<Navbar />);
      expect(shallowToJson(output)).toMatchSnapshot();
    });
});

describe('<Searchbar />', () => {
    it('should render correctly', () => {
      const output = shallow(<Searchbar />);
      expect(shallowToJson(output)).toMatchSnapshot();
    });
});

describe('<Zipcode />', () => {
    it('should render correctly', () => {
      const output = shallow(<Zipcode />);
      expect(shallowToJson(output)).toMatchSnapshot();
    });
});
