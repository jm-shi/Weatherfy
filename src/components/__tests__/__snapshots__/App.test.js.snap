// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/forecast/:zipcode"
    />
    <Route
      render={[Function]}
    />
  </Switch>
</BrowserRouter>
`;

exports[`<Home /> should render correctly 1`] = `
<div>
  <Navbar
    onHomePage="true"
  />
  <Zipcode />
</div>
`;

exports[`<Navbar /> should render correctly 1`] = `
<div
  className="header"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="logo"
    exact={true}
    to="/"
  >
    WEATHER
  </NavLink>
  <div
    style={
      Object {
        "marginTop": "20px",
      }
    }
  >
    <Searchbar />
  </div>
</div>
`;

exports[`<Searchbar /> should render correctly 1`] = `
<div
  className="search-container"
>
  <input
    id="search-bar"
    onChange={[Function]}
    placeholder="Enter zipcode"
    type="text"
    value=""
  />
  <Link
    className="button"
    onClick={[Function]}
    replace={false}
    to={
      Object {
        "pathname": "/forecast/",
        "zipcode": "",
      }
    }
  >
    Search
  </Link>
</div>
`;

exports[`<Zipcode /> should render correctly 1`] = `
<div>
  <form
    className="form"
  >
    <div
      className="form-container"
    >
      <label
        id="search-label"
      >
        Get weather by zipcode
      </label>
      <Searchbar />
    </div>
  </form>
</div>
`;
